examples-gallery/plot_parameter_estimation.py

```diff
@@ -0,0 +1,205 @@
+# %%
+"""
+
+Parameter Estimation
+====================
+
+Four noisy measurements of the location of a simple system consisting of a mass
+connected to a fixed point by a spring and a damper. The movement is in a
+horizontal direction. The the spring constant and the damping coefficient
+are to be estimated.
+
+The idea is to set up four sets of eoms, one for each of the measurements, with
+identical parameters, and let opty estimate the parameters.
+
+**State Variables**
+
+- :math:`x_1`: position of the mass of the first system [m]
+- :math:`x_2`: position of the mass of the second system [m]
+- :math:`x_3`: position of the mass of the third system [m]
+- :math:`x_4`: position of the mass of the fourth system [m]
+- :math:`u_1`: speed of the mass of the first system [m/s]
+- :math:`u_2`: speed of the mass of the second system [m/s]
+- :math:`u_3`: speed of the mass of the third system [m/s]
+- :math:`u_4`: speed of the mass of the fourth system [m/s]
+
+**Parameters**
+
+- :math:`m`: mass for both systems system [kg]
+- :math:`c`: damping coefficient for both systems [Ns/m]
+- :math:`k`: spring constant for both systems [N/m]
+- :math:`l_0`: natural length of the spring [m]
+
+"""
+# %%
+# Set up the equations of motion and integrate them to get the noisy measurements.
+#
+import sympy as sm
+import numpy as np
+import sympy.physics.mechanics as me
+import matplotlib.pyplot as plt
+from scipy.integrate import solve_ivp
+from opty import Problem
+from opty.utils import parse_free
+
+N = me.ReferenceFrame('N')
+O, P1, P2, P3, P4 = sm.symbols('O P1 P2 P3 P4', cls=me.Point)
+
+O.set_vel(N, 0)
+t = me.dynamicsymbols._t
+
+x1, x2, x3, x4 = me.dynamicsymbols('x1 x2 x3 x4')
+u1, u2, u3, u4 = me.dynamicsymbols('u1 u2 h3 u4')
+m, c, k, l0 = sm.symbols('m c k l0')
+
+P1.set_pos(O, x1 * N.x)
+P2.set_pos(O, x2 * N.x)
+P3.set_pos(O, x3 * N.x)
+P4.set_pos(O, x4 * N.x)
+P1.set_vel(N, u1 * N.x)
+P2.set_vel(N, u2 * N.x)
+P3.set_vel(N, u3 * N.x)
+P4.set_vel(N, u4 * N.x)
+
+body1 = me.Particle('body1', P1, m)
+body2 = me.Particle('body2', P2, m)
+body3 = me.Particle('body3', P3, m)
+body4 = me.Particle('body4', P4, m)
+bodies = [body1, body2, body3, body4]
+
+forces = [(P1, -k * (x1 - l0) * N.x - c * u1 * N.x),
+    (P2, -k * (x2 - l0) * N.x - c * u2 * N.x), (P3, -k * (x3 - l0) * N.x - c * u3 * N.x),
+    (P4, -k * (x4 - l0) * N.x - c * u4 * N.x)]
+
+kd = sm.Matrix([u1 - x1.diff(), u2 - x2.diff(), u3 - x3.diff(), u4 - x4.diff()])
+
+q_ind = [x1, x2, x3, x4]
+u_ind = [u1, u2, u3, u4]
+
+KM = me.KanesMethod(N, q_ind, u_ind, kd_eqs=kd)
```

Comment:
I think the example would be clearer if you didn't use KanesMethod. You can write these EoMs directly with SymPy and it is readable.

```python
Matrix([
    x1.diff(t) - u1,
    m*u1.diff(t) + c*u1 + k*(x1 - l0),
])
```